httpbp/client_middlewares.go

```diff
@@ -349,3 +355,26 @@ func PrometheusClientMetrics(serverSlug string) ClientMiddleware {
 		})
 	}
 }
+
+// ClientHeaderBPMiddleware is a middleware that forwards baseplate headers from the context to the outgoing request.
+//
+// If it detects any new baseplate headers set on the request, it will reject the request and return an error.
+func ClientHeaderBPMiddleware(client string) ClientMiddleware {
+	return func(next http.RoundTripper) http.RoundTripper {
+		return roundTripperFunc(func(req *http.Request) (*http.Response, error) {
+			for k := range req.Header {
+				if err := headerbp.CheckClientHeader(k,
+					headerbp.WithHTTPClient("", client, ""),
+				); err != nil {
+					return nil, err
+				}
```

Comment:
```suggestion
			for key := range req.Header {
				if err := headerbp.CheckClientHeader(
				  key,
					headerbp.WithHTTPClient("", client, ""),
				); err != nil {
					return nil, err
				}
```

:no_bell: I find this slightly more readable :slightly_smiling_face: